segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -1370,11 +1384,14 @@ private String getSystemJournalChunkName(int containerId, long epoch, long curre
     private <R> CompletableFuture<R> executeExclusive(Callable<CompletableFuture<R>> operation) {
         return CompletableFuture.completedFuture(null).thenComposeAsync(v -> {
             try {
+                Preconditions.checkState(exclusiveExecutionGuard.compareAndSet(false, true), "PANIC: Non exclusive execution");
```

Comment:
This looks similar to a built-in mutex (`synchronized` block); I suppose the reason you did it this way was to reject, rather than block the execution of the call until the previous call is done. How would this behave (differently) if we used a mutex here?

Reply:
unnecessary check. removing